### DISCUSSION TITLE
Where `stash` expect `ffmpeg` to be?

### DISCUSSION THREAD

**Discussion by martadinata666** [stashapp/stash#3822] [Q&A] [answered]
I had `ffmpeg` on `/usr/local/bin/ffmpeg` that usually or mostly detected correctly. I already exported the `path`, `PATH=/home/docker/.local/bin:/home/docker/.local/lib:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin`
But `stash` it can't find it, and downloading ffmpeg for no reason. Is it `/usr/bin/ffmpeg`? I saw the docker build, it normally install ffmpeg `apk add ffmpeg` that lead to `/usr/bin/ffmpeg` but stash somehow take a blind of it 🤔 

Self build ffmpeg with vaapi
```
ffmpeg version n6.0-vaapi Copyright (c) 2000-2023 the FFmpeg developers
built with gcc 12 (Debian 12.2.0-14)
configuration: --pkg-config-flags=--static --extra-version=vaapi --extra-libs='-lpthread -lm' --ld=g++ --bindir=/usr/local/bin --enable-static --disable-asm --disable-debug --disable-doc --disable-ffplay --enable-vaapi --enable-opencl --enable-gpl --enable-gnutls --enable-libaom --enable-libass --enable-libfreetype --enable-libmp3lame --enable-libopus --enable-libsvtav1 --enable-libdav1d --enable-libvorbis --enable-libvpx --enable-libx264 --enable-libx265 --enable-libvpl --enable-libsrt --disable-sdl2 --disable-libxcb --disable-xlib --enable-iconv --enable-libfdk-aac --enable-nonfree
libavutil      58.  2.100 / 58.  2.100
libavcodec     60.  3.100 / 60.  3.100
libavformat    60.  3.100 / 60.  3.100
libavdevice    60.  1.100 / 60.  1.100
libavfilter     9.  3.100 /  9.  3.100
libswscale      7.  1.100 /  7.  1.100
libswresample   4. 10.100 /  4. 10.100
libpostproc    57.  1.100 / 57.  1.100
```

Thanks.

**Comment by thevrownguy** [comment]
Its amazing to me how these questions go unanswered. I contribute cold hard cash and still cannot find goo answers sometimes. If the dev thinks discord is the answer, he is mistaken. 

Here is a great idea, why not allow the user to point to the ffmpeg installation like almost every other program. Sure you can do it the mard way but why not make it easy on the reagular people. 

If you are using it in docker you need to have all the extra stuff that nvidia requres for containers. Its a beat down. I just installed on a small virtual machine and then install cuda toolkit and the correct driver and it works great.

**Comment by WithoutPants** [comment, marked as answer]
Stash currently resolves `ffmpeg` and `ffprobe` from the `PATH` environment variable, and if it does not find it there (or if the version found there is missing some preferred flags) it searches in the same directory as the `config.yml` file. 

> Its amazing to me how these questions go unanswered. I contribute cold hard cash and still cannot find goo answers sometimes. If the dev thinks discord is the answer, he is mistaken.

There's a handful of unanswered questions on here, and most queries are answered on the Discord. I think it's expected that some things will fall through the cracks. That said, if you feel that you're not getting value from your cold hard cash, feel free to take it and your entitled attitude elswhere.

**Reply by thevrownguy** [reply]
Then why not point a person to said answwer on discord???? A link, hint... something. Oh... becasue you really cant... as discord is the most god awful way of comunicating that has ever been thought of. Its like a bunch of kids in one room yelleing over each other. The search function is deplorable and go forbid you scoll through thousands of "how do I install this questions. A kind response pointing thias user in the right direction would have been the tight way to handle this... not just ignoring it. It not like the discossions are blowing up with messages. 
I bet people would not use my product if I spoke to them the way you just did. Typical. 

It is well documented as of late how all the devs hide in discord where the yelling teenagers and CHAOS push away real people with legit answers. 
This guy did everything you in your "documentation" and he was still having an issue. Is you attitude... "I dont care... it should work.... Ill ignore it" Thats sure how it seems.

You need to hear this and I dont care if you ban me. This is not my only github account

**Reply by WithoutPants** [reply]
To be clear, the original poster did nothing wrong by posting here, and I am sorry that this question fell through the cracks. As far as I can determine the original query is answered, but if it isn't I'll leave it to OP to ask further. 

The addendum was directed at you and you only.

**Comment by martadinata666** [comment]
As 0.26 https://github.com/stashapp/stash/pull/4688 tries to resolve path handling, I guess? But I don't find it work. It is just like the previous version, that the stash binary needs the same directory as ffmpeg. So I don't know, what's going on?
```
ab64c76a6c8b% where stash 
/home/docker/.local/bin/stash
ab64c76a6c8b% where ffmpeg
/usr/local/bin/ffmpeg
ab64c76a6c8b% where ffprobe
/usr/local/bin/ffprobe
ab64c76a6c8b% echo $PATH
/home/docker/.local/bin:/home/docker/.local/lib:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
ab64c76a6c8b% 
```
